### PR TITLE
5098 branch 15 update remaining invalid engine params

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Update to use RpcErrorTypes INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS, INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMETER, INVALID_ENGINE_PAYLOAD_PARAMETER, and INVALID_ENGINE_PREPARE_PAYLOAD_PARAMETER
## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/5098

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogs.java [hunk 1]
```diff
@@ -107,12 +114,14 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
...
           .log();
-      if (ex.get() instanceof IllegalArgumentException) {
+      if (ex.get() instanceof InvalidJsonRpcParameters) {
         return new JsonRpcErrorResponse(
-            requestContext.getRequest().getId(), RpcErrorType.EXCEEDS_RPC_MAX_BLOCK_RANGE);
+            requestContext.getRequest().getId(),
+            ((InvalidJsonRpcParameters) ex.get()).getRpcErrorType());
+      } else {
+        LOG.error("Unexpected exception: {}", ex.get(), ex.get());
+        // TODO: wrap and throw? Something obviously went wrong
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogs.java:123]: think we should throw here

Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogs.java]: Yep, handled in a previous branch and now merged to main.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java [hunk 2]
```diff
@@ -80,10 +81,25 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
...
     final Object requestId = requestContext.getRequest().getId();
 
-    final EngineForkchoiceUpdatedParameter forkChoice =
-        requestContext.getRequiredParameter(0, EngineForkchoiceUpdatedParameter.class);
-    final Optional<EnginePayloadAttributesParameter> maybePayloadAttributes =
-        requestContext.getOptionalParameter(1, EnginePayloadAttributesParameter.class);
+    final EngineForkchoiceUpdatedParameter forkChoice;
+    try {
+      forkChoice = requestContext.getRequiredParameter(0, EngineForkchoiceUpdatedParameter.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
```
Comment (jframe) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:87]: Are the TODO:replace with JsonRpcParameter.JsonRpcParameterException happening in a followup PR?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java [hunk 3]
```diff
@@ -80,10 +81,25 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
...
+          e);
+    }
+    final Optional<EnginePayloadAttributesParameter> maybePayloadAttributes;
+    try {
+      maybePayloadAttributes =
+          requestContext.getOptionalParameter(1, EnginePayloadAttributesParameter.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+      throw new InvalidJsonRpcParameters(
+          "Invalid engine payload attributes parameter (index 1)",
+          RpcErrorType.INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMS,
```
Comment (jframe) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:100]: This name is similar to `INVALID_ENGINE_PAYLOAD_PARAMS` used in the new payload and it's not clear that this is a engine forkchoice updated parameter that is failing. Don't think we need a new separate error type and can reuse `INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS`. Will the Exception message `Invalid engine payload attributes parameter (index 1)` show somewhere so we can distinguish which parameter failed?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java [hunk 4]
```diff
@@ -107,8 +108,15 @@ public AbstractEngineNewPayload(
...
 
-    final EnginePayloadParameter blockParam =
-        requestContext.getRequiredParameter(0, EnginePayloadParameter.class);
+    final EnginePayloadParameter blockParam;
+    try {
+      blockParam = requestContext.getRequiredParameter(0, EnginePayloadParameter.class);
+    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+      throw new InvalidJsonRpcRequestException(
+          "Invalid engine payload parameter (index 0)",
+          RpcErrorType.INVALID_ENGINE_PAYLOAD_PARAMS,
```
Comment (jframe) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java:117]: From the name it's not clear this part of the engine newpayload RPC. Perhaps rename to `INVALID_ENGINE_NEW_PAYLOAD_PARAMS`?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java (hunk 2 above)
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:87]: Yes. It involves putting in a checked exception, which would force code changes in all of the places these PRs are updating.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java (hunk 3 above)
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:100]: The exception message will only show up in logs. The message returned in the rpc response will be the one specified in the RpcErrorType. I named it INVALID_ENGINER_PAYLOAD_ATTRIBUTES_PARAMS due to the type of object being extracted from the request (in this case `EnginePayloadAttributesParameter`). Reusing INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS in this location would mean both parameters required in AbstractEngineForkchoiceUpdated would give the same error message.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java (hunk 4 above)
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java:117]: Sure, makes sense

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java (hunk 3 above)
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:100]: Ok, I'll make that update now

**Review by jframe — APPROVED** [review]

**Pull request merged**